serializer-deserializer/src/main/java/com/amazonaws/services/schemaregistry/deserializers/protobuf/ProtobufWireFormatDecoder.java

```diff
@@ -32,37 +32,50 @@
     public Object decode(@NonNull byte[] data, @NonNull Descriptors.FileDescriptor descriptor,
                          ProtobufMessageType messageType) throws IOException {
         final CodedInputStream codedInputStream = CodedInputStream.newInstance(data);
-        Descriptors.Descriptor messageDescriptor;
-        int messageIndex;
-        try {
-            messageIndex = codedInputStream.readUInt32();
-        } catch (IOException e) {
-            throw new UncheckedIOException(e);
-        }
+        final int messageIndex = codedInputStream.readUInt32();
 
-        if (messageIndex < 0) {
```

Comment:
If removing this, where else do we check the negative message index?

Reply:
This can never be negative as we are calling`readUInt32` (unsigned int) from the buffer.